### PR TITLE
Addressed Feedback and Redesigned Profile

### PR DESCRIPTION
**Pull request by javidangarcia**
- Redesigned profile and profile history components for a cleaner and more user friendly interface.
- Fixed some code based on feedback.

https://github.com/javidangarcia/stock-zone/assets/89328228/4c131758-f61d-49dd-9445-dd3d1f7201fc

https://github.com/javidangarcia/stock-zone/assets/89328228/bd8bd4ee-08c7-4629-8370-57b51b4899dc

### CODE REVIEW

File: frontend/src/components/StocksYouFollow/StocksYouFollow.jsx [hunk 1]
```diff
@@ -112,7 +114,7 @@ export default function StocksYouFollow({ stocks, setStocks }) {
             {stocks.length > displayedStocks ? (
                 <Button
                     variant="outline-primary"
-                    onClick={() => setDisplayedStocks((prev) => prev + 3)}
+                    onClick={() => setDisplayedStocks((prev) => prev + STOCKS_TO_DISPLAY)}
```
Comment (vandanasub) [inline frontend/src/components/StocksYouFollow/StocksYouFollow.jsx:117]: 👍

File: frontend/src/components/Profile/Profile.css [hunk 2]
```diff
@@ -1,36 +1,7 @@
...
-}
-
-.profile-details h1 {
-    margin: 0;
-}
-
-.profile-details p {
-    margin: 5px 0;
-}
+.profile-header {
```
Comment (vandanasub) [inline frontend/src/components/Profile/Profile.css:5]: nice cleanup, CSS looks simpler

File: frontend/src/components/ProfileHistory/ProfileHistory.jsx [hunk 3]
```diff
@@ -3,31 +3,61 @@ import { useState, useEffect, useContext } from "react";
...
 
     useEffect(() => {
         const fetchHistory = async () => {
             try {
                 const [followsResponse, likesResponse, dislikesResponse] =
                     await Promise.all([
-                        axios.get(`${import.meta.env.VITE_HOST}/follows/user/${username}`, { withCredentials: true, validateStatus: () => true }),
-                        axios.get(`${import.meta.env.VITE_HOST}/likes/user/${username}`, { withCredentials: true, validateStatus: () => true }),
-                        axios.get(`${import.meta.env.VITE_HOST}/dislikes/user/${username}`, { withCredentials: true, validateStatus: () => true })
+                        axios.get(
```
Comment (vandanasub) [inline frontend/src/components/ProfileHistory/ProfileHistory.jsx:22]: did the linter do this?

File: frontend/src/components/ProfileHistory/ProfileHistory.jsx [hunk 4]
```diff
@@ -37,58 +67,61 @@ export default function ProfileHistory({ username }) {
...
-                                        <img src={stock.logo} alt={`This is a logo of ${stock.name}.`} />
-                                        <p>{stock.ticker}</p>
-                                    </div>
-                                </Link>
-                            </div>
-                        )
-                    })
-                }
+        <>
+            <ToggleButtonGroup
```
Comment (vandanasub) [inline frontend/src/components/ProfileHistory/ProfileHistory.jsx:71]: Nice, I like the use of the `ToggleButtonGroup` here

**Review by vandanasub — COMMENTED** [review]
Redesigned Profile page looks great!

File: frontend/src/components/ProfileHistory/ProfileHistory.jsx (hunk 3 above)
Comment (BriptimusPrimus) [reply frontend/src/components/ProfileHistory/ProfileHistory.jsx:22]: Or was it prettier? We need to make sure eslint and prettier do not contradict each other

File: frontend/.eslintrc.cjs [hunk 5]
```diff
@@ -1,15 +1,32 @@
...
+    "env": {
+        "browser": true,
+        "es2021": true
+    },
+    "extends": [
         "eslint:recommended",
-        "plugin:react/recommended",
-        "plugin:react/jsx-runtime",
-        "plugin:react-hooks/recommended"
+        "plugin:react/recommended"
```
Comment (BriptimusPrimus) [inline frontend/.eslintrc.cjs:8]: We need to make sure eslint and prettier do not contradict each other. If you see any problems, maybe try adding prettier here.
Example: 
https://github.com/BriptimusPrimus/sorted-list/blob/main/.eslintrc.js#L2

File: frontend/src/components/StockNews/StockNews.jsx [hunk 6]
```diff
@@ -86,8 +90,9 @@ export default function StockNews({ stocks }) {
             })}
             {stockNews?.length > articlesToShow ? (
                 <Button
+                    className="mb-5"
                     variant="outline-primary"
-                    onClick={() => setArticlesToShow((prev) => prev + 3)}
+                    onClick={() => setArticlesToShow((prev) => prev + ARTICLES_TO_SHOW)}
```
Comment (BriptimusPrimus) [inline frontend/src/components/StockNews/StockNews.jsx:95]: Good job! No magic numbers

File: frontend/src/utils.js [hunk 7]
```diff
@@ -112,7 +121,7 @@ export function getMarketNewsUrl() {
 export function getStockNewsUrl(currentStock) {
     const stockNewsUrl = new URL("https://finnhub.io/api/v1/company-news");
     stockNewsUrl.searchParams.append("symbol", currentStock);
-    stockNewsUrl.searchParams.append("from", "2023-07-01");
+    stockNewsUrl.searchParams.append("from", getTwoWeeksAgoDate());
```
Comment (BriptimusPrimus) [inline frontend/src/utils.js:124]: 👍

**Pull request merged**